### DISCUSSION TITLE
How to perserve typed routes with overrode router-link?

### DISCUSSION THREAD

**Discussion by aleksanderwalczuk** [posva/unplugin-vue-router#332] [Questions and Help] [answered]
Hi, I'm not sure if this case qualifies for an issue. This is my first post so I'm sorry for further inconvenience (if any). 

TLDR;
Check my example and comments at `src/pages/index.vue` and try to change `to` attribute to valid/invalid on both link components with TS support enabled.
.
https://stackblitz.com/~/github.com/aleksanderwalczuk/replace-typed-router-link

--- 

I needed to extend the default `router-link` component behaviour and set it as default. When I did that I see that `RouteNamedMap` is inferred at `:to` attribute correctly, but expected TS support has been lost.

Despite that I've created a separate component - `button-link` which inherits my (overrode) `router-link`. It's typed exactly the same, types work as expected.

What's the correct way to keep TS support on overrode `router-link`? Should this post be turned into an issue?

Thanks for your time and support.

**Comment by posva** [comment, marked as answer]
I'm not completely sure about what makes it work in one case but not the other. However, you should be able to type the `to` prop with `RouteLocationRaw` imported from `vue-router/auto`. I recommend you to check for the typed router functions like `push()` to get a better understanding of what types are typed. You can also check the client.d.ts file in the source code

**Reply by aleksanderwalczuk** [reply]
Thanks for your time, I really appreciate it. `RouteLocationRaw` solved my problem
